### ISSUE TITLE
[BUG] Garbage collector throws error in backend when creating a workspace placeholder (PHP 8)

### ISSUE THREAD

**Issue by baschny** [TYPO3-Solr/ext-solr#3641]
If you have a workspace and edit an existing record, you get an Exception when trying to save it for the first time:
```
PHP Warning: Undefined array key 69665 in /app/private/typo3conf/ext/solr/Classes/GarbageCollector.php line 227
```

**To Reproduce**
1.  Login as an user with a workspace
2. Edit some tt_content which is already published
3. Hit "Save"
4. See above mentioned exception

_Note_: The record is still saved correctly.

**Expected behavior**
No error

**Used versions:**
 - TYPO3 Version: 11.
 - EXT:solr Version: 11.5.2
 - PHP Version: 8.1

**Findings**

The problem is that the workspace "Save" will create a shadow copy of the record and `Util::isDraftRecord` is not checking for it correctly.

**Comment by baschny** [comment]
I added a fix for this in #3642 and pointed it to 11.5 branch so that it can be fixed also in the current "stable" version.

**Comment by kitzberger** [comment]
In the meantime the relevant code parts have been "fixed" by some backport to the 11.5 branch (315ddf397e3f9cd8574288282d27bc49675ee6f4). But it's not working!

@baschny's fix is though. Please merge.
